### PR TITLE
Replaced two value enum by named bool parameter

### PR DESCRIPTION
**Pull request by YohDeadfall**
As the description states this change removes two state `GenerateDirection` by a simple named bool parameter which makes code less wordy and reduces number of checks.

### CODE REVIEW

File: src/Avalonia.Layout/FlowLayoutAlgorithm.cs [hunk 1]
```diff
@@ -502,9 +501,9 @@ private bool IsReflowRequired()
 
                 // Ensure that both minor and major directions are taken into consideration so that if the scrolling direction
                 // is the same as the flow direction we still stop at the end of the viewport rectangle.
-                shouldContinue =
-                    (direction == GenerateDirection.Forward && elementMajorStart < rectMajorEnd && elementMinorStart < rectMinorEnd) ||
-                    (direction == GenerateDirection.Backward && elementMajorEnd > rectMajorStart && elementMinorEnd > rectMinorStart);
```
Comment (YohDeadfall) [inline src/Avalonia.Layout/FlowLayoutAlgorithm.cs:507]: These lines triggered me to make that change.

**Comment by grokys** [comment]
This code is ported from UWP and I was trying to make it as similar as possible to the code there in order to make it easier to port future changes from upstream. It's true that this probably won't make porting much more difficult, but I'm hesitant to change it "just because"...

**Comment by YohDeadfall** [comment]
I proposed the same change to the original source as well: microsoft/microsoft-ui-xaml#4412.

**Comment by grokys** [comment]
Great! Looks like this one got merged upstream. This PR contains additional changes though so I'll close it and open a separate PR with the changes ported from usptream.

**Pull request closed without merging**